### PR TITLE
Release Godot 4.1 beta 1 and publish release notes

### PR DESCRIPTION
**Pull request by YuriSizov**

### CODE REVIEW

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+Finally, Godot 4.1 brings back frame delta smoothing, first introduced in Godot 3.4 ([GH-52314](https://github.com/godotengine/godot/pull/52314)) by [lawnjelly](https://github.com/lawnjelly). You can read more about it in our [3.4 release blog post](/article/godot-3-4-is-released).
+
+### C#
+
+The team of .NET contributors have bridged the gap between C# and GDScript in Godot 4 in terms of feature parity. However, one glaring missing feature was global classes. In GDScript you can register any class as globally accessible with the `class_name` keyword, which, among other things, makes it appear in various parts of the editor UI. For example, you can create custom resources and instantiate them from the appropriately typed fields in the inspector dock. C# has lacked this functionality, but thanks to efforts from [Raul Santos](https://github.com/raulsntos) and [Will Nations](https://github.com/willnationsdev) (as well as rigorous testing from many other contributors) this is no longer the case ([GH-72619](https://github.com/godotengine/godot/pull/72619)).
```
Comment (raulsntos) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]: Would it be possible to add a link to the new documentation page about the global classes feature in C#? https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_global_classes.html

Comment (YuriSizov) [reply collections/_article/dev-snapshot-godot-4-1-beta-1.md]: Yes, of course!

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+![An example of interactive multi-layered RVO avoidance](/storage/blog/godot-4-1-features/rvo-avoidance.webp)
+
+To help you get started with the navigation server smix8 has prepared an [updated documentation](https://docs.godotengine.org/en/latest/tutorials/navigation/index.html) for both 2D and 3D navigation.
+
+### Rendering
+
+Earlier this year we have published a list of [rendering priorities](/article/rendering-priorities-4-1) for Godot 4.1 and future 4.x releases. While some of these tasks are still in development, a lot of time and effort has been dedicated to improving stability of various rendering backends, addressing issues, and increasing performance. One of the bigger improvements to performance in this release is newly implemented Vulkan pipeline cache ([GH-76348](https://github.com/godotengine/godot/pull/76348)) by [Alexander Streng](https://github.com/warriormaster12). It is responsible for shorter start-up times on subsequent launches of your Godot projects.
```
Comment (clayjohn) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
In April we published a list of [rendering priorities](/article/rendering-priorities-4-1) for Godot 4.1 and future 4.x releases. As expected, many of these tasks were not finished in time to be included in 4.1, but they will remain priorities for the 4.2 dev cycle and beyond if necessary. We spent more time than expected on bug fixing and investigating performance issues, but we are still quite happy with what we accomplished. One of the more notable improvements in this release is the newly implemented Vulkan pipeline cache ([GH-76348](https://github.com/godotengine/godot/pull/76348)) by [Alexander Streng](https://github.com/warriormaster12). While Godot already cached shaders to reduce shader compilation stutter, compiling pipelines still lead to some stuttering and slower load times. While the pipeline compilation stuttering issue is far from solved, this represents a step in the right direction and should also lead to a slight decrease in load times when using one of the RD-based rendering backends.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+author: Yuri Sizov
+image: /storage/blog/covers/dev-snapshot-godot-4-1-beta-1.webp
+image_caption_title: "Halls of Torment"
+image_caption_description: "A game by Chasing Carrots"
+date: 2023-06-07 18:00:00
+---
+
+After [3 months of improvements](/article/release-management-4-1) made to the engine Godot 4.1 is ready for beta testing!
+
+The main focus of this version of the engine is on gradual changes that polish user experience and stabilize features added in [Godot 4.0](/article/godot-4-0-sets-sail). At the same time, engine contributors managed to implement quite a few exciting new tools, as well as majorly rework some of the existing systems to make creating games and apps with Godot even more convenient.
```
Comment (clayjohn) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
The main focus of this release is gradual changes that polish user experience and stabilize features added in [Godot 4.0](/article/godot-4-0-sets-sail). At the same time, engine contributors managed to implement quite a few exciting new tools, as well as majorly rework some of the existing systems to make creating games and apps with Godot even more convenient.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+You may have noticed that some of these improvements have found their way into already published Godot 4.0.x releases, or may be included in future ones. If you are not ready to updated to 4.1 any time soon, we are committed to provide compatible fixes to Godot 4.0 as long as our capacity allows it.
```
Comment (clayjohn) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
You may have noticed that some of these improvements have found their way into already published Godot 4.0.x releases. If you are not ready to update to 4.1 yet, we are committed to providing compatible fixes to Godot 4.0 as long as our capacity allows it.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+### C#
+
+The team of .NET contributors have bridged the gap between C# and GDScript in Godot 4 in terms of feature parity. However, one glaring missing feature was global classes. In GDScript you can register any class as globally accessible with the `class_name` keyword, which, among other things, makes it appear in various parts of the editor UI. For example, you can create custom resources and instantiate them from the appropriately typed fields in the inspector dock. C# has lacked this functionality, but thanks to efforts from [Raul Santos](https://github.com/raulsntos) and [Will Nations](https://github.com/willnationsdev) (as well as rigorous testing from many other contributors) this is no longer the case ([GH-72619](https://github.com/godotengine/godot/pull/72619)). You can get started with this new feature by reading the [freshly updated documentation](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_global_classes.html) on global classes in Godot C#.
```
Comment (clayjohn) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
The team of .NET contributors have mostly bridged the gap between C# and GDScript in Godot 4 in terms of feature parity. However, one glaring missing feature was global classes. In GDScript you can register any class as globally accessible with the `class_name` keyword, which, among other things, makes it appear in various parts of the editor UI. For example, you can create custom resources and instantiate them from the appropriately typed fields in the inspector dock. C# has lacked this functionality, but thanks to efforts from [Raul Santos](https://github.com/raulsntos) and [Will Nations](https://github.com/willnationsdev) (as well as rigorous testing from many other contributors) this is no longer the case ([GH-72619](https://github.com/godotengine/godot/pull/72619)). You can get started with this new feature by reading the [freshly updated documentation](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_global_classes.html) on global classes in Godot C#.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+Godot 4.1 is the first in a line of releases that are aiming to aid developers in that regard. Thanks to multiple contributions from [Pedro J. Estébanez](https://github.com/RandomShaper), a big chunk of multi-threading related issues have been addressed, bringing significant improvements to the `WorkerThreadPool` class ([GH-76945](https://github.com/godotengine/godot/pull/76945), [GH-76999](https://github.com/godotengine/godot/pull/76999)) and threaded resource loading ([GH-74405](https://github.com/godotengine/godot/pull/74405), [GH-77143](https://github.com/godotengine/godot/pull/77143)). In turn, [Juan Linietsky](https://github.com/reduz) kickstarted work on an exciting new feature — scene multi-threading ([GH-75901](https://github.com/godotengine/godot/pull/75901)). It's still an experimental feature and will require additional work from all maintainers and contributors in Godot 4.2, 4.3, and further — to ensure thread safety and stability of all engine components.
```
Comment (clayjohn) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
Godot 4.1 is the first in a line of releases that are aiming to aid developers in that regard. Thanks to multiple contributions from [Pedro J. Estébanez](https://github.com/RandomShaper), a big chunk of multi-threading related issues have been addressed, bringing significant improvements to the `WorkerThreadPool` class ([GH-76945](https://github.com/godotengine/godot/pull/76945), [GH-76999](https://github.com/godotengine/godot/pull/76999)) and threaded resource loading ([GH-74405](https://github.com/godotengine/godot/pull/74405), [GH-77143](https://github.com/godotengine/godot/pull/77143)). In turn, [Juan Linietsky](https://github.com/reduz) kickstarted work on an exciting new feature — scene multi-threading ([GH-75901](https://github.com/godotengine/godot/pull/75901)). It's still an experimental feature and will require additional work from all maintainers and contributors in Godot 4.2, 4.3, and further — to ensure thread safety and stability of all engine components. To be clear, we don't recommend using scene multi-threading in existing projects just yet.
```

**Review by clayjohn — COMMENTED** [review]
Great work! This is already looking really good. 

I left a few suggestions, mostly for style/grammar fixes for some areas with awkward language and I also added a bit more explanation about the rendering priorities so users don't feel like we just dropped the priorities

File: _data/versions.yml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 - name: "4.1"
-  flavor: "dev4"
-  release_date: "1 June 2023"
-  release_notes: "/article/dev-snapshot-godot-4-1-dev-4/"
+  flavor: "beta1"
+  release_date: "7 June 2023"
```
Comment (akien-mga) [inline _data/versions.yml]:
```suggestion
  release_date: "8 June 2023"
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
+---
+title: "Dev snapshot: Godot 4.1 beta 1"
+excerpt: "Switching gears into the bug-fixing mode here's the first beta of Godot 4.1, and a great opportunity to try the new version of the engine ahead of the official release."
+categories: ["pre-release"]
+author: Yuri Sizov
+image: /storage/blog/covers/dev-snapshot-godot-4-1-beta-1.webp
+image_caption_title: "Halls of Torment"
+image_caption_description: "A game by Chasing Carrots"
+date: 2023-06-07 18:00:00
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
date: 2023-06-08 10:00:00
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
+---
+title: "Dev snapshot: Godot 4.1 beta 1"
+excerpt: "Switching gears into the bug-fixing mode here's the first beta of Godot 4.1, and a great opportunity to try the new version of the engine ahead of the official release."
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
excerpt: "Switching gears into the bug-fixing mode, here's the first beta of Godot 4.1, and a great opportunity to try the new version of the engine ahead of the official release."
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+categories: ["pre-release"]
+author: Yuri Sizov
+image: /storage/blog/covers/dev-snapshot-godot-4-1-beta-1.webp
+image_caption_title: "Halls of Torment"
+image_caption_description: "A game by Chasing Carrots"
+date: 2023-06-07 18:00:00
+---
+
+After [3 months of improvements](/article/release-management-4-1) made to the engine Godot 4.1 is ready for beta testing!
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
After [3 months of improvements](/article/release-management-4-1) made to the engine, Godot 4.1 is ready for beta testing!
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+[Jump to the **Downloads** section](#downloads), and give it a spin right now, or continue reading to learn more about changes that come in Godot 4.1. You can also [try the **Web editor**](https://editor.godotengine.org/releases/4.1.beta1/) or the **Android editor** for this release. If you are interested in the latter, please request to join [our testing group](https://groups.google.com/g/godot-testers) to get access to pre-release builds.
+
+*The illustration picture for this article is from* [**Halls of Torment**](https://store.steampowered.com/app/2218750/Halls_of_Torment/), *a roguelite action RPG with retro aesthetics calling back to late-90s pre-rendered 2D graphics. It is developed by [Chasing Carrots](https://twitter.com/chasing_carrots) using Godot 4.0, and is available in [early access on Steam](https://store.steampowered.com/app/2218750/Halls_of_Torment/), so you can buy it right now!*
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
*The illustration picture for this article is from* [**Halls of Torment**](https://store.steampowered.com/app/2218750/Halls_of_Torment/), *a roguelite action RPG with retro aesthetics calling back to late-90s pre-rendered 2D graphics. It is developed by [Chasing Carrots](https://twitter.com/chasing_carrots) using Godot 4.0, and it just released in [early access on Steam](https://store.steampowered.com/app/2218750/Halls_of_Torment/), so you can buy it right now! Congrats to the team for the overwhelmingly positive reception!*
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+Another area of C# development that received attention during this release cycle is configurability. For example, [Alex de la Mare](https://github.com/alexdlm) provided an option for fine-grained disabling of source generators ([GH-71049](https://github.com/godotengine/godot/pull/71049)) and [RedworkDE](https://github.com/RedworkDE) made include scripts customizable on the export preset level instead of a global setting ([GH-72896](https://github.com/godotengine/godot/pull/72896)). This later features comes with a bigger improvement beyond just .NET, as all editor export plugins can now define custom export options for users to provide ([GH-72895](https://github.com/godotengine/godot/pull/72895)).
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]: I'm not sure I understand "This later features comes with...", is this grammatically correct? Was it meant to be "The latter features come with..." (i.e. "The features we last mentioned come with...").

Might need rephrasing either way to be easier to understand.

**Review by akien-mga — COMMENTED** [review]
Partial review up to "Editor" (included), I'll continue and finish in 20 min.

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md (hunk 12 above)
Comment (YuriSizov) [reply collections/_article/dev-snapshot-godot-4-1-beta-1.md]: Should be "latter feature", of course! But let me think about rephrasing it all together.

Comment (akien-mga) [reply collections/_article/dev-snapshot-godot-4-1-beta-1.md]: "The latter feature comes with" might be fine already. I mostly felt the need to rephrase if this had to be plural and refer the the last X mentioned features which would be a bit of an unconventional construct.

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+The functionality of GDExtensions have also been improved. [Yuri Rubinsky](https://github.com/Chaosus) made it possible to define new visual shader nodes with extensions ([GH-70911](https://github.com/godotengine/godot/pull/70911)), while David implemented the framework for custom editor plugins ([GH-77010](https://github.com/godotengine/godot/pull/77010)). And if you register custom nodes with your extensions, you can now give them fitting individual icons thanks to a contribution from [Yuri Sizov](https://github.com/YuriSizov) ([GH-75472](https://github.com/godotengine/godot/pull/75472)).
+
+### GDScript
+
+After a big refactoring and a plethora of new features that GDScript received in Godot 4.0 it was time for polish and internal clean-up, squashing inconsistencies and bugs. One notable improvement to an existing feature comes from [ocean](https://github.com/anvilfolk) who took a deep dive into the script documentation generation and emerged with a solid rework ([GH-72095](https://github.com/godotengine/godot/pull/72095)). Among other things, this rework significantly improves how enumeration types are treated in the generated documentation, making them proper named types.
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]: I believe commas are needed after introductory clauses like "After [timed event], [consequence]":
https://www.grammarly.com/blog/commas-after-introductory-phrases/
```suggestion
After a big refactoring and a plethora of new features that GDScript received in Godot 4.0, it was time for polish and internal clean-up, squashing inconsistencies and bugs. One notable improvement to an existing feature comes from [ocean](https://github.com/anvilfolk) who took a deep dive into the script documentation generation and emerged with a solid rework ([GH-72095](https://github.com/godotengine/godot/pull/72095)). Among other things, this rework significantly improves how enumeration types are treated in the generated documentation, making them proper named types.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+Experienced Godot users have long noted that while GDScript supported static methods, it was impossible to preserve any sort of state from such methods due to lack of static class variables. In Godot 3 some developers devised tricks using the fact that a constant Dictionary is mutable or relied on object metadata to work around this limitation. Unfortunately, many such tricks became impossible in Godot 4 as a result of language improvements and bug fixes. Good news is that starting Godot 4.1 GDScript officially supports static class variables, enabling new programming patterns and bringing a much needed quality-of-life change, thanks to [George Marques](https://github.com/vnen).
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
Experienced Godot users have long noted that while GDScript supported static methods, it was impossible to preserve any sort of state from such methods due to lack of static class variables. In Godot 3 some developers devised tricks using the fact that a constant Dictionary is mutable or relied on object metadata to work around this limitation. Unfortunately, many such tricks became impossible in Godot 4 as a result of language improvements and bug fixes. The good news is that starting Godot 4.1, GDScript officially supports static class variables, enabling new programming patterns and bringing a much needed quality-of-life change, thanks to [George Marques](https://github.com/vnen).
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+### Multi-threading
+
+A major goal of Godot 4 is modernization of the engine core and engine's systems — to serve as a foundation for many years of development and further improvements. With a decade of advancements in CPU core and thread count it is essential for modern software to be able to utilize these available hardware resources. A strong multi-threading support allows developers to better scale their games, both single-player and multiplayer, and provide a smoother, more performant experience to their users.
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
A major goal of Godot 4 is the modernization of the engine core and engine's systems — to serve as a foundation for many years of development and further improvements. With a decade of advancements in CPU core and thread count, it is essential for modern software to be able to utilize these available hardware resources. A strong multi-threading support allows developers to better scale their games, both single-player and multiplayer, and provide a smoother, more performant experience to their users.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md (hunk 12 above)
Comment (YuriSizov) [reply collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
Another area of C# development that received attention during this release cycle is configurability. For example, [Alex de la Mare](https://github.com/alexdlm) provided an option for fine-grained disabling of source generators ([GH-71049](https://github.com/godotengine/godot/pull/71049)) and [RedworkDE](https://github.com/RedworkDE) made include scripts customizable on the export preset level ([GH-72896](https://github.com/godotengine/godot/pull/72896)), where previously you would have a global project setting. In fact, this last feature comes with a positive change for anyone creating editor export plugins, as they now have a new API to define custom export options for users to supply ([GH-72895](https://github.com/godotengine/godot/pull/72895)).
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+
+### Rendering
+
+In April we published a list of [rendering priorities](/article/rendering-priorities-4-1) for Godot 4.1 and future 4.x releases. As expected, many of these tasks were not finished in time to be included in 4.1, but they will remain priorities for the 4.2 dev cycle and beyond if necessary. We spent more time than expected on bug fixing and investigating performance issues, but we are still quite happy with what we accomplished. One of the more notable improvements in this release is the newly implemented Vulkan pipeline cache ([GH-76348](https://github.com/godotengine/godot/pull/76348)) by [Alexander Streng](https://github.com/warriormaster12). While Godot already cached shaders to reduce shader compilation stutter, compiling pipelines still lead to some stuttering and slower load times. While the pipeline compilation stuttering issue is far from solved, this represents a step in the right direction and should also lead to a slight decrease in load times when using one of the RD-based rendering backends.
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
In April we published a list of [rendering priorities](/article/rendering-priorities-4-1) for Godot 4.1 and future 4.x releases. As expected, many of these tasks were not finished in time to be included in 4.1, but they will remain priorities for the 4.2 dev cycle and beyond if necessary. We spent more time than expected on bug fixing and investigating performance issues, but we are still quite happy with what we accomplished. One of the more notable improvements in this release is the newly implemented Vulkan pipeline cache ([GH-76348](https://github.com/godotengine/godot/pull/76348)) by [Alexander Streng](https://github.com/warriormaster12). While Godot already cached shaders to reduce shader compilation stutter, compiling pipelines still lead to some stuttering and slower load times. While the pipeline compilation stuttering issue is far from solved, this represents a step in the right direction and should also lead to a slight decrease in load times when using one of the RenderingDevice-based rendering backends.
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+Particles are another part of the visual tech behind the engine that has received a lot of attention in this release. A new particle turbulence system, fully compatible with existing projects, have been implemented to empower technical artists to create impressive and beautiful dynamic effects ([GH-64606](https://github.com/godotengine/godot/pull/64606), [GH-77154](https://github.com/godotengine/godot/pull/77154)). It took a bit of deliberation and time, but [KdotJPG](https://github.com/KdotJPG) and [Raffaele Picca](https://github.com/RPicster) were able to get everything just right and the team cannot wait to see your upcoming creations!
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion

Particles are another part of the visual tech behind the engine that has received a lot of attention in this release. The particle turbulence system originally introduced in 4.0 has been reworked to better empower technical artists to create impressive and beautiful dynamic effects ([GH-64606](https://github.com/godotengine/godot/pull/64606), [GH-77154](https://github.com/godotengine/godot/pull/77154)). It took a bit of deliberation and time, but [KdotJPG](https://github.com/KdotJPG) and [Raffaele Picca](https://github.com/RPicster) were able to get everything just right and the team cannot wait to see your upcoming creations!
```

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+
+To help even more with setting up the atmosphere of your 3D scenes contributors [Clay John](https://github.com/clayjohn), [Johan](https://github.com/JohanAR), and [Patrick](https://github.com/paddy-exe) implemented new and fixed existing shader built-ins ([GH-71364](https://github.com/godotengine/godot/pull/71364), [GH-76109](https://github.com/godotengine/godot/pull/76109), [GH-76290](https://github.com/godotengine/godot/pull/76290)). Last but not least, [Lasuch](https://github.com/Lasuch69) and Clay worked on the new 3D noise texture ([GH-76486](https://github.com/godotengine/godot/pull/76486), [GH-76557](https://github.com/godotengine/godot/pull/76557)) which can be used to add more depth to your volumetric fog or to affect the behavior of 3D particles. Quite the first contribution to the engine for Lasuch!
```
Comment (akien-mga) [inline collections/_article/dev-snapshot-godot-4-1-beta-1.md]:
```suggestion
To help even more with setting up the atmosphere of your 3D scenes, contributors [Clay John](https://github.com/clayjohn), [Johan](https://github.com/JohanAR), and [Patrick](https://github.com/paddy-exe) implemented new and fixed existing shader built-ins ([GH-71364](https://github.com/godotengine/godot/pull/71364), [GH-76109](https://github.com/godotengine/godot/pull/76109), [GH-76290](https://github.com/godotengine/godot/pull/76290)). Last but not least, [Lasuch](https://github.com/Lasuch69) and Clay worked on the new 3D noise texture ([GH-76486](https://github.com/godotengine/godot/pull/76486), [GH-76557](https://github.com/godotengine/godot/pull/76557)) which can be used to add more depth to your volumetric fog or to affect the behavior of 3D particles. Quite the first contribution to the engine for Lasuch!
```

**Review by akien-mga — COMMENTED** [review]
Reviewed the second half.

**Review by akien-mga — APPROVED** [review]
Really good writeup!

I commented on some minor style issues but once fixed this should be ready to publish.

File: collections/_article/dev-snapshot-godot-4-1-beta-1.md (hunk 13 above)
Comment (YuriSizov) [reply collections/_article/dev-snapshot-godot-4-1-beta-1.md]: This is my Achilles' heel when it comes to English! In Russian you practically always add a comma after introductory clauses, or really anywhere where you would have a [pause] while speaking. But in English many of those are dropped, and I always get confused 🙃

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**